java/src/main/java/scenarios/crewcreation/CharacterCreation.java

```diff
@@ -0,0 +1,247 @@
+        // TODO: Implement functionality of the rollAbilityScore method
+
+        int meanScore = (int) unassignedAbilityScores.stream().mapToInt(i -> i).average().orElse(0);
+
+        System.out.println("Looks like these are your ability scores: " + unassignedAbilityScores);
+
+        if (meanScore == 18) {
+            System.out.println("What?!?!?");
+            System.out.println("There's no way...");
+            System.out.println("I can't believe it! You've truly won the genetic lottery!");
+        } else if (meanScore >= 14) {
+            System.out.println("Very impressive! You remind me of myself back in my younger days...");
+        } else if (meanScore >= 10) {
+            System.out.println("Too good to be bad, but too bad to be good. You must have something to prove.");
+        } else if (meanScore == 9) {
+            System.out.println("I must say, you're exceptionally average.");
+            System.out.println("You're like a ruler for the rest of the universe to measure themselves against...");
+        } else if (meanScore >= 7) {
+            System.out.println("You're a bit below average, but I guess you've never let that stop you before...");
+        } else if (meanScore >= 4) {
+            System.out.println("These stats are quite low... life must be a bit of a struggle for you, huh?");
+        } else {
+            System.out.println("Wow... I don't think I've ever seen anything quite so terrible before.");
+            System.out.println("Just so you know, there's no shame in restarting the game.");
+            System.out.println("I probably would, if I were you...");
+        }
+
+        // TODO: (A) Have player assign their ability scores
+
+        Integer[] abilityScoreValues = askAbilityScores(unassignedAbilityScores);
+
+        int physical = abilityScoreValues[0];
+        int agility = abilityScoreValues[1];
+        int blaster = abilityScoreValues[2];
+        int piloting = abilityScoreValues[3];
+        int repair = abilityScoreValues[4];
+        int charisma = abilityScoreValues[5];
+
+
+        //return null;
+        /* TODO: (A)
+        The above line is here so you can run the dice tests, otherwise the compiler would complain about using values
+        that haven't been set. Delete the line above and uncomment the line below when you get the dice tests passing
+        and you're ready to start adding the functionality for the player to set their ability scores using the values
+        that they rolled.
+        */
+        return new Captain(name, physical, agility, blaster, piloting, repair, charisma);
+    }
+
+    public static void printAbilityScores(List<Integer> list) {
+        for (int i = 0; i < list.size(); i++) {
+            System.out.println("[" + (i + 1) + "]" + " " + list.get(i));
+        }
+    }
+
+    public static Integer[] askAbilityScores(List<Integer> list) {
+        Scanner scanner = new Scanner(System.in);
```

Comment:
You can delete this line since we have a scanner as a field of this class now, hence we no longer need to create an instance of it within any of the methods.